lib/lightning_web/live/dataclip_live/index.html.heex

```diff
@@ -1,52 +1,61 @@
-<MainSection.header title={@page_title} />
+<MainSection.header title={@page_title}>
+  <%= live_redirect to: Routes.project_dataclip_edit_path(@socket, :new, @project.id) do %>
+    <Common.button>
+      <div class="h-full">
+        <Icon.plus class="h-4 w-4 inline-block" />
+        <span class="inline-block align-middle">New Dataclip</span>
+      </div>
+    </Common.button>
+  <% end %>
+</MainSection.header>
 
 <MainSection.main>
   <%= if @live_action == :show do %>
     <.show_dataclip {assigns} />
   <% end %>
 
   <%= if @live_action == :index do %>
-    <div class="grid grid-cols-10 gap-4 font-bold">
-      <div class="col-span-2">
-        Created
-      </div>
-      <div class="col-span-2">Type</div>
-      <div class="col-span-4">Body</div>
+    <.table>
+      <.tr>
+        <.th>Created</.th>
+        <.th>Type</.th>
+        <.th>Body</.th>
+        <.th>Actions</.th>
+      </.tr>
 
-      <div class="col-span-2">Action</div>
-    </div>
-    <%= for dataclip <- @page.entries do %>
-      <div class="grid grid-cols-10 gap-4" id={"dataclip-#{dataclip.id}"}>
-        <div class="col-span-2">
-          <%= dataclip.inserted_at |> Calendar.strftime("%c") %>
-        </div>
-        <div class="col-span-2"><.type_pill dataclip={dataclip} /></div>
-        <div class="col-span-4"><%= dataclip.body %></div>
+      <%= for dataclip <- @page.entries do %>
+        <.tr id={"dataclip-#{dataclip.id}"}>
+          <.td>
+            <%= dataclip.inserted_at |> Calendar.strftime("%c") %>
```

Comment:
I don't have a solid alternative right now (that isn't cludgy), but in liveview templates function calls should be avoided: https://hexdocs.pm/phoenix_live_view/assigns-eex.html#change-tracking

Reply:
ok, let's leave for now but i hear you.